device/tt_silicon_driver.cpp

```diff
@@ -2243,7 +2243,8 @@ void tt_SiliconDevice::read_from_non_mmio_device(void* mem_ptr, tt_cxy_pair core
             } else {
                 // Read 4 byte aligned block from device/sysmem
                 if (use_dram) {
-                    read_from_sysmem(data_block, host_dram_block_addr, host_dram_channel, block_size, mmio_capable_chip_logical);
+                    size_buffer_to_capacity(data_block, block_size);
```

Comment:
Somewhat unrelated to your change, but pointing it out as something I noticed --

Is `data_block` necessary, or could sysmem/device memory be copied to from the source pointer (`mem_ptr`) directly?  I created https://github.com/tenstorrent/tt-umd/issues/90

Reply:
Thanks for pointing it out, @broskoTT and I are working on prototype of the new API on the branch https://github.com/tenstorrent/tt-umd/tree/brosko/new_api_experiment, so it will require rethinking remote API. That will be a good chance to take a look at this stuff